### ISSUE TITLE
Oracle view creation fails while execute the SQL script via Flyway

### ISSUE THREAD

**Issue by Sajith1988V** [flyway/flyway#2048]
###### Which version and edition of Flyway are you using?

Using Flyway Flyway Community Edition 5.1.1 by Boxfuse

###### If this is not the latest version, can you reproduce the issue with the latest one as well?
(Many bugs are fixed in newer releases and upgrading will often resolve the issue)

###### Which client are you using? (Command-line, Java API, Maven plugin, Gradle plugin)
Command-line

###### Which database are you using (type & version)?
Oracle 12.2.1, Ojdbc 8

###### Which operating system are you using?
Windows & Unix

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, the SQL statement that failed (if relevant) and the command you ran.)

CREATE OR REPLACE VIEW PA_SUMMARY_VIEW AS
    WITH
        FUNCTION Test_funct (
            CL_ID   IN VARCHAR2,
            P_DIN   IN VARCHAR2
        ) RETURN NUMBER AS
            V_O_AMT   NUMBER(8,2);
        BEGIN
.
.
.
.
This is a new feature of Oracle 12 C to allow functions/procedures inside a with clause. I am trying to create a view out of this with clause query. This executes successfully when I run the script from SQL* Plus or SQL Developer. But this fails when executed via Flyway 5.1.1 Open source

###### What did you expect to see?

View created successfully. 

###### What did you see instead?

Encountered the below error

**SQL State  : 65000
Error Code : 6553
Message    : ORA-06553: PLS-103: Encountered the symbol "end-of-file" when expecting one of the following:

   := ; not null default character**

**Comment by axelfontaine** [comment]
Please share the full statement Flyway fails to parse correctly, so we can make sure you case is covered.

**Comment by Sajith1988V** [comment]
Here is the full view query:



CREATE OR REPLACE VIEW PA_SUMMARY_VIEW AS
    WITH
-- Total Amount Function    
        FUNCTION WITH_TOT_AMT (
            CL_ID   IN VARCHAR2,
            P_DIN   IN VARCHAR2
        ) RETURN NUMBER AS
            V_O_AMT   NUMBER(8,2);
        BEGIN
            SELECT
                SUM(A.TOTAL_AMT_PAID) TOTAL_AMT_$
            INTO V_O_AMT
            FROM
                table1 A,
                table2 B
            WHERE
                A.CARRIER_ID = 15
                AND A.CEXP_REC_ID = 4
                AND A.CLAIM_STATUS = 0
                AND A.DATE_XREF_NUM = 0
                AND A.CLIENT_ID = CL_ID
                AND A.DIN = P_DIN
                AND A.DATE_PROCESSED <> TO_CHAR(SYSDATE,'yyyymmdd')
                AND A.DATE_PROCESSED = B.DATE_PROCESSED
                AND A.CARRIER_ID = B.CARRIER_ID
                AND A.CLAIM_REF_NUM = B.CLAIM_REF_NUM
                AND A.TRACE_NUM = B.TRACE_NUM;

            RETURN V_O_AMT;
        END;

-- Total quantity function 

        FUNCTION WITH_TOT_QTY (
            CL_ID   IN VARCHAR2,
            P_DIN   IN VARCHAR2
        ) RETURN NUMBER IS
            V_O_QTY   NUMBER(8,2);
        BEGIN
            SELECT
                SUM(C.MQTY_PAID) TOTAL_QTY
            INTO V_O_QTY
            FROM
                table1 A,
                table2 C
            WHERE
                A.CARRIER_ID = 15
                AND A.CEXP_REC_ID = 4
                AND A.CLAIM_STATUS = 0
                AND A.DATE_XREF_NUM = 0
                AND A.CLIENT_ID = CL_ID
                AND A.DIN = P_DIN
                AND A.DATE_PROCESSED <> TO_CHAR(SYSDATE,'yyyymmdd')
                AND A.DATE_PROCESSED = C.DATE_PROCESSED
                AND A.CARRIER_ID = C.CARRIER_ID
                AND A.CLAIM_REF_NUM = C.CLAIM_REF_NUM
                AND A.TRACE_NUM = C.TRACE_NUM;


            RETURN V_O_QTY;
        END;

--Main query 

    SELECT
        PA_NUMBER,
        CLIENT_ID,
		DIN,
        PA_DATE,
        APPROVED_QTY,
        APPROVED_AMOUNT_$,
        DECODE(SIGN(APPROVED_QTY - TOT_QTY),-1,0,APPROVED_QTY - NVL(TOT_QTY,0) ) OUTSTANDING_QTY,
        DECODE(SIGN(APPROVED_AMOUNT_$ - TOT_AMT),-1,0,APPROVED_AMOUNT_$ - NVL(TOT_AMT,0) ) OUTSTANDING_AMT_$
    FROM
        (
            SELECT
                A.PA_NUMBER,
                A.PA_DATE,
                NVL(B.PA_DIN,' ') DIN,
                NVL(B.PA_QTY,0) APPROVED_QTY,
                NVL(B.PA_TOTAL_AMT,0) APPROVED_AMOUNT_$,
                A.CLIENT_ID,
                WITH_TOT_QTY(A.CLIENT_ID,B.PA_DIN) TOT_QTY,
                WITH_TOT_AMT(A.CLIENT_ID,B.PA_DIN) TOT_AMT
            FROM
                tt1 A,
                tt2 B
            WHERE
                A.CARRIER_ID = 15
                AND A.GROUP_NUM = '080000'
                 AND A.PA_FUNCTION <> '   '
                AND B.CLIENT_ID (+) = A.CLIENT_ID
                AND B.PA_NUMBER (+) = A.PA_NUMBER

        );
        /

**Comment by Sajith1988V** [comment]
Please keep us posted once this is fixed in the next milestone.

**Comment by axelfontaine** [comment]
I can't seem to find documentation for this syntax anywhere, including at https://docs.oracle.com/en/database/oracle/oracle-database/12.2/sqlrf/CREATE-VIEW.html

Can you point me to the official docs regarding this?

**Comment by axelfontaine** [comment]
Any news?

**Comment by Sajith1988V** [comment]
This view was created while executed via sql plus and sql developer.

**Comment by axelfontaine** [comment]
But where is this documented as official Oracle syntax though?

**Comment by Sajith1988V** [comment]
https://asktom.oracle.com/pls/asktom/f?p=100:11:0::::P11_QUESTION_ID:9534452300346807610

**Comment by Sajith1988V** [comment]
Any updates???

**Comment by apexbine** [comment]
I'm interested, too!
Here's another Link: https://oracle-base.com/articles/12c/with-clause-enhancements-12cr1#functions

Are there any workarounds for this? (Something more elegant than execute immediate ...)

**Comment by apexbine** [comment]
By the way, @Sajith1988V, the date for the milestone 5.2.0 is August 28, 2018.

**Comment by Sajith1988V** [comment]
Hi, 
Is there a possibility to expedite the fix for this issue?

**Comment by Sajith1988V** [comment]
Please provide which version of flyway has this fix. I am unable to find any latest version.

**Comment by axelfontaine** [comment]
@Sajith1988V We only offer pre-release binaries to paying Pro or Enterprise customers.

If this is not an option for you, you can simply build from source yourself or wait until the next release.

**Comment by Sajith1988V** [comment]
Ok Axel, I will try to build from my source. 
By the way may I know when is the next release planned for?

**Comment by Sajith1988V** [comment]
HI Axel, 
May I know if the fix for this is released in a new version? I do not see any latest updates to download.

**Comment by pputerla** [comment]
Hi @axelfontaine ,

I've noticed that you have effectively disabled CE/Pro support for Oracle 12.1 (https://github.com/flyway/flyway/commit/d915b860ebcf27003b51efc76618af0ab6f331bd)

Was that deliberately?
I was wondering - do you have some policy on that? I mean we'd only like to know  what to expect.
You know what I mean...

**Comment by Sajith1988V** [comment]
@pputerla, 
Is this the reason the view creation failed for me?

**Comment by axelfontaine** [comment]
@pputerla Yes, that is deliberate. See https://github.com/flyway/flyway/issues/2147

The policy is: Oracle version still covered by Oracle Premier Support?
- Yes: Supported in all editions of Flyway
- No: Supported in Flyway Enterprise Edition only

@Sajith1988V No.

**Comment by pputerla** [comment]
#2147: That's what I call a policy :) thanks!

**Comment by Sajith1988V** [comment]
@axelfontaine when is the next version of flyway be released that supports my view creation syntax? Please expedite. I do not see any upgraded version yet to download. 

You had replied me a month back saying 

"We only offer pre-release binaries to paying Pro or Enterprise customers.

If this is not an option for you, you can simply build from source yourself or wait until the next release."

**Comment by jakubek2085** [comment]
Thanks to this fix, I cannot migrate a view which does not contain WITH FUNCTION|PROCEDURE,  but a CTE (common table expression). Needs fixing.

**Comment by pauxus** [comment]
Yes, we encountered the same problem right now as well.

The regexp needs to be less specific:

```java
"^CREATE(\\s+OR\\s+REPLACE)?(\\s+(NON)?EDITIONABLE)?\\s+(VIEW)\\s+.*\\s+AS\\s+WITH\\s+(FUNCTION|PROCEDURE).*");
```
